packages/runtime/container-runtime/src/runningSummarizer.ts

```diff
@@ -216,6 +230,64 @@ export class RunningSummarizer implements IDisposable {
 		});
 	}
 
+	public async handleSummaryAck(): Promise<number> {
+		const lastAck: IAckedSummary | undefined = this.summaryCollection.latestAck;
+		let refSequenceNumber = -1;
+		if (lastAck !== undefined) {
+			refSequenceNumber = lastAck.summaryOp.referenceSequenceNumber;
+			const summaryLogger = this.tryGetCorrelatedLogger(refSequenceNumber) ?? this.logger;
+			try {
+				const summaryOpHandle = lastAck.summaryOp.contents.handle;
+				const summaryAckHandle = lastAck.summaryAck.contents.handle;
+				// Make sure we block any summarizer from being executed/enqueued while
+				// executing the refreshLatestSummaryAck.
+				// https://dev.azure.com/fluidframework/internal/_workitems/edit/779
+				await this.lockedRefreshSummaryAckAction(async () =>
+					this.refreshLatestSummaryAckCallback({
```

Comment:
We also call `refreshLatestSummaryAckFromServer` in the container runtime method `submitSummary`. So if we are refreshing in that scenario and then we start another summarize, we can be in trouble.

Reply:
Refresh within the summary iterations should be fine. I suppose the only concern is the call within HandleSummaryAcks